app/components/Views/confirmations/hooks/useMaxValueRefresher.ts

```diff
@@ -27,15 +34,19 @@ export function useMaxValueRefresher() {
     transactionMetadata as TransactionMeta,
   );
   const { balanceWeiInHex } = useAccountNativeBalance(chainId, txParams.from);
+  const layer1GasFee = transactionMetadata?.layer1GasFee as Hex;
 
   useEffect(() => {
-    if (!maxValueMode || type !== TransactionType.simpleSend) {
+    if (!maxModeEnabled || type !== TransactionType.simpleSend) {
       // Not compatible with transaction value refresh logic
       return;
     }
 
     const balance = new BigNumber(balanceWeiInHex);
-    const maxPossibleFee = new BigNumber(maxFeeNativeHex as Hex);
+    const maxPossibleFee = new BigNumber(maxFeeNativeHex as Hex).plus(
+      new BigNumber(layer1GasFee ?? '0x0'),
+    );
```

Comment:
```suggestion
    const maxPossibleFee = new BigNumber(maxFeeNativeHex as Hex);
```

This is already added in calculations : https://github.com/MetaMask/metamask-mobile/blob/main/app/components/Views/confirmations/hooks/gas/useFeeCalculations.ts#L147